Sources/MapboxDirectionsCLI/ProcessCommand.swift

```diff
@@ -0,0 +1,95 @@
+
+import Foundation
+import MapboxDirections
+import SwiftCLI
+
+
+private let BogusCredentials = DirectionsCredentials(accessToken: "pk.feedCafeDadeDeadBeef-BadeBede.FadeCafeDadeDeed-BadeBede")
+
+class ProcessCommand<ResponceType : Codable, OptionsType : DirectionsOptions > : Command {
+    var name = "process"
+    
+    @Key("-i", "--input", description: "Filepath to the input JSON.")
+    var inputPath: String?
+    
+    @Key("-c", "--config", description: "Filepath to the JSON, containing serialized Options data.")
+    var configPath: String?
+    
+    @Key("-o", "--output", description: "[Optional] Output filepath to save the conversion result. If no filepath provided - will output to the shell.")
+    var outputPath: String?
+    
+    var customShortDescription: String = ""
+    var shortDescription: String {
+        return customShortDescription
+    }
+    
+    
+    private func processResponse<T>(_ decoder: JSONDecoder, type: T.Type, from data: Data) throws -> Data where T : Codable {
+        let result = try decoder.decode(type, from: data)
+        let encoder = JSONEncoder()
+        return try encoder.encode(result)
+    }
+    
+    private func absURL ( _ path: String ) -> URL {
+        // some methods cannot correctly expand '~' in a path, so we'll do it manually
+        let homeDirectory = URL(fileURLWithPath: NSHomeDirectory())
+        guard path != "~" else {
+            return homeDirectory
+        }
+        guard path.hasPrefix("~/") else { return URL(fileURLWithPath: path)  }
+
+        var relativePath = path
+        relativePath.removeFirst(2)
```

Comment:
`NSString.expandingTildeInPath` is a more robust way to resolve the tilde.

Reply:
I as sure there was a reason I didn't use this method, but I can't remember it now, and it seems to work here. So I'll use it, thanks